--- ./nowplaying/src/nowplaying.cpp.old	2010-08-04 18:43:12.000000000 +0400
+++ ./nowplaying/src/nowplaying.cpp	2010-08-05 13:08:16.000000000 +0400
@@ -44,7 +44,6 @@
     }
     connect(m_oscar_protocol, SIGNAL(accountCreated(qutim_sdk_0_3::Account*)), this, SLOT(accountCreated(qutim_sdk_0_3::Account*)));
     connect(m_jabber_protocol, SIGNAL(accountCreated(qutim_sdk_0_3::Account*)), this, SLOT(accountCreated(qutim_sdk_0_3::Account*)));
-    connect(m_mrim_protocol, SIGNAL(accountCreated(qutim_sdk_0_3::Account*)), this, SLOT(accountCreated(qutim_sdk_0_3::Account*)));
 
     m_stop_start_action = new StopStartActionGenerator(this,
                                                        (m_global_settings.is_working) ? tr("Stop now playing") : tr("Start now playing"));
@@ -78,13 +77,10 @@
     QString proto_id;
     QString acc_id;
     m_accounts.insert("oscar", "icq");
-    m_accounts.insert("mrim", "mrim");
     m_accounts.insert("jabber", "jabber");
     foreach (Protocol* proto, allProtocols()) {
         if (proto->id() == "icq"){
             m_oscar_protocol = proto;
-        } else if (proto->id() == "mrim"){
-            m_mrim_protocol = proto;
         } else if (proto->id() == "jabber"){
             m_jabber_protocol = proto;
         }
@@ -114,18 +110,6 @@
             Account* account = m_oscar_protocol->account(acc);
             connect(account, SIGNAL(statusChanged(qutim_sdk_0_3::Status)), this, SLOT(statusChanged(qutim_sdk_0_3::Status)));
             connect(account, SIGNAL(destroyed()), this, SLOT(accountDeleted()));
-        } else if (proto_id == "mrim"){
-            Config saved_config = group.group(acc);
-            MRIM config;
-            config.deactivated = saved_config.value("deactivated", false);
-            config.sets_current_status = saved_config.value("sets_current_status", false);
-            config.sets_music_status = saved_config.value("sets_music_status", true);
-            config.mask_1 = saved_config.value("mask_1", QString("Now playing: %artist - %title"));
-            config.mask_2 = saved_config.value("mask_2", QString("%artist - %title"));
-            m_mrim_accs.insert(acc, config);
-            Account* account = m_mrim_protocol->account(acc);
-            connect(account, SIGNAL(statusChanged(qutim_sdk_0_3::Status)), this, SLOT(statusChanged(qutim_sdk_0_3::Status)));
-            connect(account, SIGNAL(destroyed()), this, SLOT(accountDeleted()));
         } else if (proto_id == "jabber"){
             Config saved_config = group.group(acc);
             Jabber config;
@@ -294,19 +278,6 @@
         m_current_status_name.insert(acc_id, acc->status().extendedInfo("xstatus").value("name").toString());
         connect(acc, SIGNAL(statusChanged(qutim_sdk_0_3::Status)), this, SLOT(statusChanged(qutim_sdk_0_3::Status)));
         connect(acc, SIGNAL(destroyed()), this, SLOT(accountDeleted()));
-    } else if (proto_id == "mrim"){
-        MRIM conf;
-        Config mrim = group.group(acc_id);
-        conf.deactivated = mrim.value("deactivated", false);
-        conf.sets_current_status = mrim.value("sets_current_status", false);
-        conf.sets_music_status = mrim.value("sets_music_status", true);
-        conf.mask_1 = mrim.value("mask_1", QString("Now playing: %artist - %title"));
-        conf.mask_2 = mrim.value("mask_2", QString("%artist - %title"));
-        m_mrim_accs.insert(acc_id, conf);
-        if (isAccountNeedsInSettingStatus(acc_id) && m_player->isPlaying())
-            setMRIMStatus(acc_id, getMessage(acc_id, m_player->trackInfo()));
-        connect(acc, SIGNAL(statusChanged(qutim_sdk_0_3::Status)), this, SLOT(statusChanged(qutim_sdk_0_3::Status)));
-        connect(acc, SIGNAL(destroyed()), this, SLOT(accountDeleted()));
     } else if (proto_id == "jabber"){
         Jabber conf;
         Config jabber = group.group(acc_id);
@@ -331,8 +302,6 @@
     m_accounts.remove(acc->id());
     if (proto_id == "icq"){
         m_oscar_accs.remove(acc_id);
-    } else if (proto_id == "mrim"){
-        m_mrim_accs.remove(acc_id);
     } else if (proto_id == "jabber"){
         m_jabber_accs.remove(acc_id);
     }
@@ -349,14 +318,6 @@
                 replace("%file", info.file_name).replace("%uri", info.uri).
                 replace("%time", info.time);
         return message;
-    } else if (proto_id == "mrim"){
-        MRIM config = (m_global_settings.for_all_accs) ? m_mrim_accs.value("mrim") : m_mrim_accs.value(acc);
-        QString message = (config.sets_current_status) ? config.mask_1 : config.mask_2;
-        message.replace("%artist", info.artist).replace("%title", info.title).
-                replace("%album", info.album).replace("%track", info.track_number).
-                replace("%file", info.file_name).replace("%uri", info.uri).
-                replace("%time", info.time);
-        return message;
     } else {
         return QString();
     }
@@ -412,21 +373,6 @@
         } else if (proto_id == "jabber"){
             Jabber config = (m_global_settings.for_all_accs) ? m_jabber_accs.value("jabber") : m_jabber_accs.value(acc);
             return !config.deactevated;
-        } else if (proto_id == "mrim"){
-            MRIM config = (m_global_settings.for_all_accs) ? m_mrim_accs.value("mrim") : m_mrim_accs.value(acc);
-            if (!config.deactivated){
-                if (config.sets_current_status){
-                    //complete!!!
-                    return true;
-                } else if (config.sets_music_status){
-                    //complete!!!
-                    return true;
-                } else {
-                    return false;
-                }
-            } else {
-                return false;
-            }
         }
     } else {
         return false;
